pkg/collector/corechecks/snmp/checkconfig/config.go

```diff
@@ -410,13 +420,11 @@ func NewCheckConfig(rawInstance integration.Data, rawInitConfig integration.Data
 	c.InstanceTags = instance.Tags
 	c.MetricTags = instance.MetricTags
 
-	c.OidConfig.addScalarOids(parseScalarOids(c.Metrics, c.MetricTags))
-	c.OidConfig.addColumnOids(parseColumnOids(c.Metrics))
+	c.addUptimeMetric()
 
-	if c.CollectDeviceMetadata {
-		c.OidConfig.addScalarOids(metadata.ScalarOIDs)
-		c.OidConfig.addColumnOids(metadata.ColumnOIDs)
-	}
+	c.Metadata = updateMetadataDefinitionWithLegacyFallback(nil)
+	c.OidConfig.addScalarOids(c.parseScalarOids(c.Metrics, c.MetricTags, c.Metadata))
+	c.OidConfig.addColumnOids(c.parseColumnOids(c.Metrics, c.Metadata))
```

Comment:
Why not loading the profile, if needed, before setting oids ? From my understanding here you parse the configuration, add the uptime metric, add global metrics, and add them to oids. But you don't load the profile.

Reply:
What I would expect is:
* load metrics from config
* load metrics from profile, if profile is explicitly defined or if it should be autodetected
* add uptime and global metrics
* setup oidconfig
Why are profile's metrics and their oids loaded separately ?

Reply:
> Why not loading the profile, if needed, before setting oids ?
I'm not sure to fully understand. Is the question about why we are loading all profiles as done here https://github.com/DataDog/datadog-agent/blob/b6eb373a201be993f559baf5dfb84d6ceab074ed/pkg/collector/corechecks/snmp/checkconfig/config.go#L418-L434 instead of just loading necessary profiles ?

If so, the reason is that is was simpler (probably opinionated), "just load everything, put in cache and use when needed". Performance wise, it should be fine, since it's done only once. Memory wise, it might help to only load the profile that is needed.

---

> load metrics from profile, if profile is explicitly defined or if it should be autodetected

note: the autodetection is done at check run phase, not at configuration phase.
the reasons is that, it might be possible that the device is not yet available and we want to try multiple time (at each check run) to autodetect.